diff --git a/src/misc/lv_color.h b/src/misc/lv_color.h
index 2cc92f277..9025f8712 100644
--- a/src/misc/lv_color.h
+++ b/src/misc/lv_color.h
@@ -164,7 +164,7 @@ enum {
 # define LV_COLOR_GET_A32(c) (c).ch.alpha
 
 # define _LV_COLOR_ZERO_INITIALIZER32  {{0x00, 0x00, 0x00, 0x00}}
-# define LV_COLOR_MAKE32(r8, g8, b8) {{b8, g8, r8, 0xff}} /*Fix 0xff alpha*/
+# define LV_COLOR_MAKE32(r8, g8, b8) {{r8, g8, b8, 0xff}} /*Fix 0xff alpha*/
 
 /*---------------------------------------
  * Macros for the current color depth
@@ -223,9 +223,9 @@ typedef union {
 
 typedef union {
     struct {
-        uint8_t blue;
-        uint8_t green;
         uint8_t red;
+        uint8_t green;
+        uint8_t blue;
         uint8_t alpha;
     } ch;
     uint32_t full;
